recipes-bsp/u-boot/README.md

```diff
@@ -71,3 +71,73 @@ Then upgrade the firmware:
 ```shell
 flashrom -p ch341a_spi -c W25Q128.V -w iot2050-image-boot.bin
 ```
+
+## eMMC RPMB otp key provisioning
+
+To utilize the eMMC RPMB as the backend for secure storage, an otp key is
+required to be programmed into the eMMC RPMB. This is a per-device key, each
+device has its unique key.
+
+A special firmware build is required to run in a secure operating environment to
+program this key into RPMB. To build this special firmware:
+
+```shell
+./kas-container build kas-iot2050-boot-pg2.yml:kas/opt/rpmb-setup.yml
+```
+
+This will build a special OPTee binary for generating and programming the otp
+key, together with a special U-Boot binary for kicking off the key programming.
+
+> :warning:
+> This firmware must be run in a secure environment to protect the programmed
+> otp key.
+>
+> After programming this otp key, secure boot must be enabled to protect the
+> key. Otherwise this key could be leaked from some unauthorized OPTee running
+> on the device.
+>
+> This special firmware build must not be distribute outside of the secure
+> operating environment, even with the secure boot enabled. This is due to when
+> OPTee checks if the RPMB key is programmed , it uses a supplicant to
+> communicate with the non secure world eMMC driver (since the hardware resides
+> in non-secure world to begin with). If someone replaces the supplicant with a
+> rogue application always replying "RPMB key not programmed", OP-TEE will then
+> send the key in clear text, asking for the supplicant to program it, which
+> could lead to the leakage of the key.
```

Comment:
The recommendation should be:
 - program RPMB key using UNSIGNED special firmware in secure operating environment
 - flash SIGNED regular image
 - program secure boot keys and enable secure boot

Reply:
And: NEVER sign RPMB setup firmware version.

Reply:
The whole warning reworded to:

> :warning:
> This special firmware must NOT be signed.
>
> This special firmware must run in a secure operating environment.
>
> It is recommended that after programming the RPMB otp key, flash the signed
> regular images then program the secure boot keys and enable secure boot.

Reply:
Maybe even "To protected the RPMB key, it is required to flash a signed image, program secure boot keys and enable secure boot."